### PR TITLE
Update Gradle and Plugins

### PR DESCRIPTION
**Pull request by AndrewReitz**
Updated Gradle to 4.3
Updated Kotlin to 1.1.51
Updated Android Plugin to 3.0.0

### CODE REVIEW

File: src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt [hunk 1]
```diff
@@ -101,17 +101,17 @@ abstract class TaskProvider(
         val variants: DomainObjectCollection<out BaseVariant> = when {
             project.plugins.hasPlugin("com.android.application") -> {
                 val ext = project.extensions.findByType(AppExtension::class.java)
-                ext.applicationVariants
+                ext!!.applicationVariants
```
Comment (AndrewReitz) [inline src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt]: Updating to Gradle 4.2 caused the build to fail thinking this could be nullable.

**Comment by jaredsburrows** [comment]
This PR conflicts with: https://github.com/KeepSafe/dexcount-gradle-plugin/pull/226.

**Comment by benjamin-bader** [comment]
Thanks for the contribution!  This is failing because our Travis job wasn't accepting the right SDK license.  That's fixed as of #226, which also has some overlap with this PR.

This looks fine to me; if you can rebase and fix the merge conflict, I'll be happy to land it!

**Comment by AndrewReitz** [comment]
Ok, should be good to go.

File: src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt (hunk 1 above)
Comment (jaredsburrows) [reply src/main/kotlin/com/getkeepsafe/dexcount/Plugin.kt]: Can you use `?.`?

File: gradle/versions.gradle [hunk 2]
```diff
@@ -11,7 +11,7 @@ ext {
 
 // Plugins
 ext.plugs = [
-    "gradle"              : "com.android.tools.build:gradle:3.0.0-beta6",
+    "gradle"              : "com.android.tools.build:gradle:3.0.0",
```
Comment (jaredsburrows) [inline gradle/versions.gradle]: Can you undo this change because we are going most likely use 3.1 from this PR: https://github.com/KeepSafe/dexcount-gradle-plugin/pull/225.

Comment (AndrewReitz) [reply gradle/versions.gradle]: done

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.3-bin.zip
```
Comment (jaredsburrows) [inline gradle/wrapper/gradle-wrapper.properties]: Can you make sure this stays on `-all`. Also can you run `gradlew wrapper` after you do to update the property and jar files?

**Comment by jaredsburrows** [comment]
Looks good but I can't merge this, @benjamin-bader can.

**Comment by benjamin-bader** [comment]
Thanks, @AndrewReitz - glad to have your contribution!
